src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/sender/LogSender.java

```diff
@@ -94,9 +105,13 @@ public void sendLogs(String url, byte[] msg, String sumoName, String sumoCategor
     }
 
     private void createHeaders(final PostMethod post, final String sumoName,
-                               final String sumoCategory, final String contentType) {
-
-        post.addRequestHeader("X-Sumo-Host", getHost());
+                               final String sumoCategory, final String contentType,
+                               HashMap<String, String> fields, final String host) {
+        if (StringUtils.isNotEmpty(host)) {
+            post.addRequestHeader("X-Sumo-Host", host);
+        } else {
+            post.addRequestHeader("X-Sumo-Host", getHost());
+        }
 
         if (StringUtils.isNotBlank(sumoName)) {
```

Comment:
Does isNotBlank consider null values?

Reply:
Yes. IsBlank checks for null values also.